src/canvas.jl

```diff
@@ -103,14 +103,17 @@ function lines!(c::Canvas, X::AbstractVector, Y::AbstractVector; color::UserColo
 end
 
 
-function get_canvas_dimensions_for_matrix(canvas::Type{T}, nrow::Int, ncol::Int, maxwidth::Int, maxheight::Int, width::Int, height::Int, margin::Int, padding::Int; extra_rows::Int = 0, extra_cols::Int = 0) where {T <: Canvas}
+function get_canvas_dimensions_for_matrix(
+    canvas::Type{T}, nrow::Int, ncol::Int, maxwidth::Int, maxheight::Int, width::Int, height::Int, margin::Int, padding::Int, os::Union{Nothing,IO};
```

Comment:
Urghhh... what does `os` stand for, "output stream"?  Maybe rename to [`out_stream`](https://github.com/JuliaLang/julia/blob/8812c5cc242435bf267285fcf3e41e632c84e52c/stdlib/REPL/src/Terminals.jl#L103)?

Reply:
Allright.

Reply:
https://github.com/JuliaPlots/UnicodePlots.jl/pull/157